### DISCUSSION TITLE
INAV ported to RP2040 Pico

### DISCUSSION THREAD

**Discussion by synersignart** [iNavFlight/inav-configurator#1818] [Ideas]
this is fairly recent no older than two years and a friend noted me if possible to port INAV to run on RP2040 and how far are the feature functional on this chipset. I would be interested as well to try out on a RPi Pico board to test out  a target. 

https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html

**Comment by akhodeir** [comment]
I am also interested about using INAV for rp2040.

It is rich in perphirals and has 2 cores. The flash can be extended to 16 MBytes.
one of the down sides of rp2040 that it doesn't have float point unit. I am not sure how critical is that for flight controllers. 

if an expert tell us what is required, I would like to invest more time to support this activity.

**Comment by sensei-hacker** [comment]
See also
https://github.com/iNavFlight/inav/discussions/10401
